integrations/integrations.go

```diff
@@ -0,0 +1,232 @@
+) error {
+	g, ctx := getCtx(svrCtx, true)
+
+	svrCtx.Logger.Info("starting Monomer node in-process")
+	// --- Start the Monomer node ---
+	wrappedApp := &WrappedApplication{app}
+	_, err := startMonomerNode(wrappedApp, env, svrCtx)
+	if err != nil {
+		return err
+	}
+
+	if opts.PostSetup != nil {
+		if err := opts.PostSetup(svrCtx, clientCtx, ctx, g); err != nil {
+			return err
+		}
+	}
+
+	return g.Wait()
+}
+
+// Starts the Monomer node in-process in place of the Comet node. The
+// `MonomerGenesisPath` flag must be set in viper before invoking this.
+func startMonomerNode(wrappedApp *WrappedApplication, env *environment.Env, svrCtx *server.Context) (*node.Node, error) {
+	// --- Bootstrap listeners and DBs ---
+	chainID := monomer.ChainID(1)
+	engineWS, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		return nil, err
+	}
+
+	cmtListenAddr := svrCtx.Config.RPC.ListenAddress
+	cmtListenAddr = strings.TrimPrefix(cmtListenAddr, "tcp://")
+	cometListener, err := net.Listen("tcp", cmtListenAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	appdb := dbm.NewMemDB()
+	blockdb := dbm.NewMemDB()
+	txdb := cometdb.NewMemDB()
+	mempooldb := dbm.NewMemDB()
+
+	// --- Load Monomer genesis doc ---
+	monomerGenesisPath := viper.GetString("monomer-genesis-path")
+	svrCtx.Logger.Info("loading Monomer genesis from", "path", monomerGenesisPath)
+
+	appGenesis, err := genutiltypes.AppGenesisFromFile(monomerGenesisPath)
+	if err != nil {
+		svrCtx.Logger.Error("failed to load Monomer genesis", "error", err)
+		return nil, err
+	}
+
+	var appState map[string]json.RawMessage
+	if err := json.Unmarshal(appGenesis.AppState, &appState); err != nil {
+		svrCtx.Logger.Error("failed to unmarshal app state", "error", err)
+		return nil, fmt.Errorf("failed to unmarshal app state: %w", err)
+	}
+
+	// --- Create a Monomer node ---
+	n := node.New(
+		wrappedApp,
+		&genesis.Genesis{
+			ChainID:  chainID,
+			AppState: appState,
+		},
+		engineWS,
+		cometListener,
+		blockdb,
+		mempooldb,
+		txdb,
+		&node.SelectiveListener{
+			OnEngineHTTPServeErrCb: func(err error) {
+				svrCtx.Logger.Error("failed to serve engine HTTP", "error", err)
+			},
+			OnEngineWebsocketServeErrCb: func(err error) {
+				svrCtx.Logger.Error("failed to serve engine websocket", "error", err)
+			},
+			OnCometServeErrCb: func(err error) {
+				svrCtx.Logger.Error("failed to serve CometBFT", "error", err)
+			},
+		},
+	)
+	svrCtx.Logger.Info("spinning up Monomer node")
+
+	nodeCtx, nodeCtxCancel := context.WithCancel(context.Background())
+
+	// --- Run Monomer ---
+	nodeErr := n.Run(nodeCtx, env)
+	if nodeErr != nil {
+		svrCtx.Logger.Error("failed to run Monomer node", "error", err)
+	}
+
+	// --- Check if CometBFT listener is up ---
+	if conn, err := net.Dial("tcp", cmtListenAddr); err != nil {
+		svrCtx.Logger.Error("failed to dial CometBFT", "error", err)
+	} else {
+		svrCtx.Logger.Info("dialing CometBFT succeeded at", "address", cmtListenAddr)
+		conn.Close()
+	}
+
```

Comment:
This is unnecessary, correct?

Reply:
Yeah this is unnecessary. I added it for testing but will remove it now